### PR TITLE
Do not delete ticks before counters

### PR DESCRIPTION
**Pull request by graetzer**
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/1550/

### CODE REVIEW

File: arangod/RocksDBEngine/RocksDBSettingsManager.cpp [hunk 1]
```diff
@@ -80,14 +84,11 @@ arangodb::Result writeCounterValue(
   rocksdb::Status s =
       rtrx->Put(RocksDBColumnFamily::definitions(), key.string(), value);
   if (!s.ok()) {
-    rtrx->Rollback();
-    auto rStatus = convertStatus(s);
-    LOG_TOPIC(WARN, Logger::ENGINES)
-        << "writing counters failed: " << rStatus.errorMessage();
```
Comment (danielhlarkin) [inline arangod/RocksDBEngine/RocksDBSettingsManager.cpp]: Please don't remove the log message here. If anything, replace it with one with more info, e.g.
```
LOG_TOPIC(WARN, Logger::ENGINES)
    << "writing counter for collection with objectId '" << pair.first << "' failed: " << rStatus.errorMessage();
```

File: arangod/RocksDBEngine/RocksDBSettingsManager.cpp [hunk 2]
```diff
@@ -123,11 +124,7 @@ arangodb::Result writeSettings(rocksdb::Transaction* rtrx, VPackBuilder& b,
       rtrx->Put(RocksDBColumnFamily::definitions(), key.string(), value);
 
   if (!s.ok()) {
-    auto rStatus = arangodb::rocksutils::convertStatus(s);
-    LOG_TOPIC(WARN, Logger::ENGINES)
-        << "writing settings failed: " << rStatus.errorMessage();
```
Comment (danielhlarkin) [inline arangod/RocksDBEngine/RocksDBSettingsManager.cpp]: Please don't remove the log message here.

File: arangod/RocksDBEngine/RocksDBSettingsManager.cpp [hunk 3]
```diff
@@ -80,14 +84,11 @@ arangodb::Result writeCounterValue(
   rocksdb::Status s =
       rtrx->Put(RocksDBColumnFamily::definitions(), key.string(), value);
   if (!s.ok()) {
-    rtrx->Rollback();
-    auto rStatus = convertStatus(s);
-    LOG_TOPIC(WARN, Logger::ENGINES)
-        << "writing counters failed: " << rStatus.errorMessage();
-    return rStatus;
+    return std::make_pair(convertStatus(s), returnSeq);
```
Comment (danielhlarkin) [inline arangod/RocksDBEngine/RocksDBSettingsManager.cpp]: Is it correct to return `returnSeq = baseSeq` here? Don't we need to keep `pair.second._sequenceNum` around still?

File: arangod/RocksDBEngine/RocksDBSettingsManager.cpp [hunk 4]
```diff
@@ -391,17 +388,21 @@ Result RocksDBSettingsManager::sync(bool force) {
...
+    rocksdb::SequenceNumber returnSeq;
+    
+    std::tie(res, returnSeq) = writeCounterValue(_syncedSeqNums, rtrx.get(),
+                                                 b, pair, seqNumber);
     if (res.fail()) {
       return res;
     }
 
-    auto writeResult =
+    std::tie(res, returnSeq) =
```
Comment (danielhlarkin) [inline arangod/RocksDBEngine/RocksDBSettingsManager.cpp]: Doesn't this overwrite the return from `writeCounterValue`?

File: arangod/RocksDBEngine/RocksDBSettingsManager.cpp (hunk 3 above)
Comment (graetzer) [reply arangod/RocksDBEngine/RocksDBSettingsManager.cpp]: no thats the point, otherwise we never make any progress

Comment (danielhlarkin) [reply arangod/RocksDBEngine/RocksDBSettingsManager.cpp]: Isn't this equivalent to ignoring the error and pretending we wrote out the value? Can't we end up with incorrect counters this way?

Comment (graetzer) [reply arangod/RocksDBEngine/RocksDBSettingsManager.cpp]: the code in `RocksDBSettingsManager::sync` will abort the entire operation, or am I wrong ?

Comment (danielhlarkin) [reply arangod/RocksDBEngine/RocksDBSettingsManager.cpp]: Ah, yes, you're right. So essentially the return seq value doesn't actually matter here.

**Pull request merged**